azkaban-common/src/main/java/azkaban/executor/ExecutionController.java

```diff
@@ -323,6 +324,21 @@ public long getQueuedFlowSize() {
     return size;
   }
 
+  @Override
+  public long getAgedQueuedFlowSize() {
+    long size = 0L;
+    int minimum_age_minutes = this.azkProps.getInt(
```

Comment:
Are invalid values for this (such a negative) being checked anywhere in the stack ?

Reply:
There are a lot of property where a non negative number will be expected, and supporting this within the props file will be more appropriate I think. Not having this in props is a shortcoming in the current system, for e.g:

```
final int connectionRequestTimeout =
        props.getInt(JOBCALLBACK_CONNECTION_REQUEST_TIMEOUT, DEFAULT_TIME_OUT_MS);
    final int connectionTimeout = props.getInt(JOBCALLBACK_CONNECTION_TIMEOUT, DEFAULT_TIME_OUT_MS);
    final int socketTimeout = props.getInt(JOBCALLBACK_SOCKET_TIMEOUT, DEFAULT_TIME_OUT_MS);
    this.responseWaitTimeoutMS =
        props.getInt(JOBCALLBACK_RESPONSE_WAIT_TIMEOUT, DEFAULT_RESPONSE_WAIT_TIME_OUT_MS);
```

There are no positivity or non-negativity checks for our props.We can introduce methods in props file such as:

getPositiveInt()
getNonNegativeInt().

What do you think?

Reply:
I would think that for the timeout related configs negative values are acceptable and would typically imply `no-timeout`
For most other 'time' based configs typically the underlying libraries will take care of looking for illegal arguments e.g In this case `scheduleAtFixedRate` has an explicit check for non-positive values  internally.
 ```    public void start() {
      this.scheduler.scheduleAtFixedRate(() -> pollExecution(), 0L, this.pollingIntervalMs,
          TimeUnit.MILLISECONDS);
    }
```
In your specific usage things will fail silently with the db simply not returning any values. 
A simple check within this method would be fine for this. More exhaustive solutions like modifying the `props` are likely not needed and can be taken up in future if really required.

Reply:
I have made the change. But I still wanted to post a clarification  

"> In your specific usage things will fail silently with the db simply not returning any values."

Values would have been returned. Let's go through an example to see what would have happened:

A negative value will have the same effect as a zero value.
Let's say the 
age = -50.
current_time = 800
cutoff = current_time - age = 800 - (-50) = 850
For the SQL query, the WHERE clause for submit_time will become:
(submit_time < 850).

The query wouldn't have failed. It would have returned valid results. Now, Semantic of negative age is wierd (same as negative timeouts or many other thresholds), but that wouldn't have made the query return no results.

Reply:
You are right, for this query the result set would not be empty, whether it's valid would depend on semantics we want and documented accordingly (at least in code)
Thanks for the clarification.